src/env/node/git/git.ts

```diff
@@ -1814,11 +1814,17 @@ export class Git {
 				);
 				if (data.trim() === 'true') {
 					// If we are in a bare clone, then the common dir is the git dir
-					data = await this.git<string>(
-						{ cwd: cwd, errors: GitErrorHandling.Ignore },
-						'rev-parse',
-						'--git-common-dir',
-					);
+					// We can assume rev_parse__git_dir will always return a value since we know we are in a repo
+					({ commonPath: data } = (await this.rev_parse__git_dir(cwd)) as {
+						path: string;
+						commonPath?: string;
+					});
+
+					// If the git dir is .git or a path pointed to by gitdir: ...,
+					// we should treat it as an "empty" version of a bare clone
+					// This is the case when git dir == git common dir, or when commonPath is undefined
+					data = data ?? '.';
+
```

Comment:
Please replace this with the following (assuming it works in your testing). In my testing this seems to resolve the issue and takes advantage of the path normalization that `rev_parse__git_dir` is already doing. It also avoids returning the `cwd` if `rev_parse__git_dir` really returns no data (which shouldn't happen), but keeps this safer.

```suggestion
					const result = await this.rev_parse__git_dir(cwd);
					const repoPath = result?.commonPath ?? result?.path;
					if (repoPath?.length) return [true, repoPath];
```

Reply:
The change I checked in 4017c82beb2f0e5cad4c1a83071a4452a8fa8487 (on `main` now), will allow GL to properly collapse the extra repos in most cases.

Making any further improvements/changes should be handled outside of this PR

Reply:
Alright. I tested the changes and it works great. Thank you so much!